plugins/modules/fabric_devices_workflow_manager.py

```diff
@@ -0,0 +1,4080 @@
+                self.delete_l2_handoff(have_l2_handoff, device_ip,
+                                       result_fabric_device_response,
+                                       result_fabric_device_msg).check_return_status()
+
+            have_sda_l3_handoff = have_fabric_device.get("sda_l3_handoff_details")
+            if item.get("sda_l3_handoff_details"):
+                self.delete_sda_l3_handoff(have_sda_l3_handoff, device_ip,
+                                           result_fabric_device_response,
+                                           result_fabric_device_msg).check_return_status()
+
+            have_ip_l3_handoff = have_fabric_device.get("ip_l3_handoff_details")
+            if item.get("ip_l3_handoff_details"):
+                self.delete_ip_l3_handoff(have_ip_l3_handoff, device_ip,
+                                          result_fabric_device_response,
+                                          result_fabric_device_msg).check_return_status()
+            delete_fabric_device = have_fabric_device.get("delete_fabric_device")
+            device_exists = have_fabric_device.get("exists")
+
+            # If the delete_fabric_device is set to True
+            # We need to delete the device as well along with the settings
+            if delete_fabric_device:
+                if device_exists:
+                    id = have_fabric_device.get("id")
+                    try:
+                        payload = {"id": id}
+                        task_name = "delete_fabric_device_by_id"
+                        task_id = self.get_taskid_post_api_call("sda", task_name, payload)
+                        if not task_id:
+                            self.msg = (
+                                "Unable to retrive the task_id for the task '{task_name}'."
+                                .format(task_name=task_name)
+                            )
+                            self.set_operation_result("failed", False, self.msg, "ERROR")
+                            return self
+
+                        success_msg = (
+                            "Successfully deleted the SDA fabric device with IP '{ip}'."
+                            .format(ip=device_ip)
+                        )
+                        self.get_task_status_from_tasks_by_id(task_id, task_name, success_msg).check_return_status()
+                        result_fabric_device_msg.update({
+                            "device_details": "SDA device successfully removed from fabric."
+                        })
+                    except Exception as msg:
+                        self.msg = (
+                            "Exception occured while deleting the fabric device with IP '{ip}': {msg}"
+                            .format(ip=device_ip, msg=msg)
+                        )
+                        self.log(self.msg, "ERROR")
+                        self.status = "failed"
+                        return self
+                else:
```

Comment:
self.log("Fabric device with IP '{device_ip}' not found in Cisco Catalyst Center.".format(device_ip=device_ip), "INFO")